polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -987,12 +987,36 @@ fn sanitize_backed_candidate_v2<T: crate::inclusion::Config>(
 	allowed_relay_parents: &AllowedRelayParentsTracker<T::Hash, BlockNumberFor<T>>,
 	allow_v2_receipts: bool,
 ) -> bool {
+	// Get the claim queue snapshot at the candidate relay parent.
+	let Some((rp_info, _)) =
+		allowed_relay_parents.acquire_info(candidate.descriptor().relay_parent(), None)
+	else {
+		log::debug!(
+			target: LOG_TARGET,
+			"Relay parent {:?} for candidate {:?} is not in the allowed relay parents.",
+			candidate.descriptor().relay_parent(),
+			candidate.candidate().hash(),
+		);
+		return false
+	};
+
+	if let Err(err) = candidate.candidate().check_ump_signals(&rp_info.claim_queue) {
```

Comment:
We should skip this check if candidate is v2 and v2 not allowed by features.

Reply:
It's checked right after. I don't think the order matters in this case

Reply:
You can just skip and not process the UMP signals needlessly.